### PR TITLE
[Feature] Create new project from the project dropdown

### LINKED ISSUE #3101: [Feature] Task | Create new project in Add Task or Task Detail page
**Related Document:** 
- [Document 1](https://www.notion.so/everco/As-a-team-member-I-want-to-create-a-new-project-directly-from-the-task-add-edit-form-when-the-proje-114dfdd17ffd80429791e5d1c72c9146?pvs=4)
- [Document 2](https://www.notion.so/everco/As-a-team-member-with-the-Users-Team-Members-permissions-I-should-be-able-to-add-tasks-in-the-proje-10ddfdd17ffd80509759d061377c98e2)

**Related issues:**
- #3098
- #3099 
- #3100 

---

As a team member, I want to create a new project directly from the task add/edit form when the project I need isn't listed, so I can efficiently categorize my task without interrupting my workflow.

**Detailed Flow:**
1. User is in the process of adding or editing a task
2. User clicks on the project dropdown
3. User scans the list but doesn't find the desired project
4. User selects add button at the bottom of the dropdown
5. System opens a quick project creation modal
6. User enters essential project details (project name and color)
7. User submits the new project information
8. System creates the new project and adds it to the user's project list
9. The project dropdown updates, showing the newly created project as selected
10. User continues with task creation/editing

**Acceptance Criteria:**
1. Project Dropdown:
    * An add button is clearly visible in the project dropdown
    * This option is typically placed at the bottom of the project list
2. Quick Project Creation:
    * Selecting add button opens a modal
    * The form includes fields for essential project information 
3. User Experience:
    * The project creation process is quick and doesn't navigate away from the task add/edit page
    * Form for new project is simple and doesn't require extensive information to create
4. Project Creation and Selection:
    * Upon successful creation, the new project is immediately available in the dropdown
    * The newly created project is automatically selected for the current task
    
**Additional Information:**
image

image
Add project should have similat button like this

image
Show a modal like this when creating new project

### PR DESCRIPTION
**Pull request by CREDO23**
## Description

Fix #3101 

## Type of Change

-   [ ] Bug fix
-   [x] New feature
-   [ ] Breaking change
-   [ ] Documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [x] My changes generate no new warnings

## Current screenshots

[Loom](https://www.loom.com/share/b7df1a81031e4b7096feb3e916b0c4aa?sid=3ddb75c5-a3a7-4d7b-b954-cc82e6ace02f)

## Summary by CodeRabbit

- **New Features**
	- Introduced a modal for creating new projects within the task details section.
	- Enhanced the project dropdown with improved navigation, detailed project options, and functionality to create new projects.

- **Improvements**
	- Updated the `ProjectIcon` component for dynamic sizing based on props.
	- Enhanced the `TaskStatusDropdown` for better usability and responsiveness.
	- Added localization support for the "Create New" action across multiple languages.

- **Bug Fixes**
	- Adjusted dropdown options to prevent fixed height issues, allowing for flexible content display.

### CODE REVIEW

File: apps/web/components/pages/task/details-section/blocks/task-secondary-info.tsx [hunk 1]
```diff
@@ -395,96 +333,115 @@
...
+												<div className="flex flex-col gap-2.5 w-full p-4">
+													{organizationProjects.map((item, i) => {
+														return (
+															<Listbox.Option key={item.id} value={item} as={Fragment}>
+																<li className="relative border h-[2rem] flex items-center gap-1 px-2 rounded-lg outline-none cursor-pointer">
+																	<ProjectIcon width={14} height={14} />{' '}
+																	<span className=" truncate">{item.name}</span>
+																</li>
+															</Listbox.Option>
+														);
```
[the remark this answers is not in the document]
Comment (CREDO23) [reply apps/web/components/pages/task/details-section/blocks/task-secondary-info.tsx:413]: that is how it should be used (read headless docs!)

File: apps/web/components/pages/task/details-section/blocks/task-secondary-info.tsx [hunk 2]
```diff
@@ -395,96 +333,115 @@
...
+					value={selected}
+					onChange={(project) => {
+						if (controlled && onChange) {
+							onChange(project);
+						} else {
+							handleUpdateProject(project);
+						}
+
+						setSelected(project);
+					}}
```
[the remark this answers is not in the document]
Comment (CREDO23) [reply apps/web/components/pages/task/details-section/blocks/task-secondary-info.tsx:354]: In both cases, we should sync both parent and children state for better UI updates, `onChange` is going to carry the parent state and `seSelected` is going to do that for the children

**Review by Cedric921 — COMMENTED** [review]
Please Resolve EN and ES translations for "CREATE_NEW" key
@CREDO23

File: apps/web/locales/ru.json [hunk 3]
```diff
@@ -267,7 +267,8 @@
 		"FILTER_CUSTOM_RANGE": "Пользовательский диапазон",
 		"CLEAR_FILTER": "Очистить фильтр",
 		"CLEAR": "Очистить",
-		"APPLY_FILTER": "Применить фильтр"
+		"APPLY_FILTER": "Применить фильтр",
+		"CREATE_NEW": "Создать новый"
```
[the remark this answers is not in the document]
Comment (CREDO23) [reply apps/web/locales/ru.json:271]: Oh, got it

File: apps/web/components/pages/task/details-section/blocks/task-secondary-info.tsx [hunk 4]
```diff
@@ -104,23 +106,6 @@ const TaskSecondaryInfo = () => {
 				</ActiveTaskVersionDropdown>
 			</TaskRow>
 
-			{/* Epic */}
```
Comment (evereq) [inline apps/web/components/pages/task/details-section/blocks/task-secondary-info.tsx:107]: @CREDO23 why this whole block is removed!?

File: apps/web/components/pages/task/details-section/blocks/task-secondary-info.tsx [hunk 5]
```diff
@@ -179,34 +145,6 @@ const TaskSecondaryInfo = () => {
 				</ActiveTaskStatusDropdown>
 			</TaskRow>
 
-			{/* Task Labels */}
-			<TaskRow labelTitle={t('pages.taskDetails.LABELS')}>
```
Comment (evereq) [inline apps/web/components/pages/task/details-section/blocks/task-secondary-info.tsx:183]: @CREDO23 why this whole block is removed?

File: apps/web/components/pages/task/details-section/blocks/task-secondary-info.tsx (hunk 4 above)
Comment (CREDO23) [reply apps/web/components/pages/task/details-section/blocks/task-secondary-info.tsx:107]: It was duplicated @evereq 

Screenshot from 2024-11-27 19-28-35

File: apps/web/components/pages/task/details-section/blocks/task-secondary-info.tsx (hunk 5 above)
Comment (CREDO23) [reply apps/web/components/pages/task/details-section/blocks/task-secondary-info.tsx:183]: This one was duplicated as well @evereq 

Screenshot from 2024-11-27 19-31-24

**Pull request merged**